### DISCUSSION TITLE
ModuleNotFoundError: No module named 'httpx' while importing FastAPI Test Client.

### DISCUSSION THREAD

**Discussion by caner-cetin** [fastapi/fastapi#9286] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

https://github.com/canercetin-randomguy/damacansu-sepeti/blob/master/api/endpoints/database/test_users_db.py#L119

### Description

First of all, thanks for this wonderful project. I dont know if I can commit to help with my dumbness, like I did linked project for learning Python/FastAPI but I will try.

I tried to import

`from fastapi.testclient import TestClient`

and called

` response = test_http_client.post("/user/signup", json=user_details.dict())`

then I got traceback when I run the test: https://pastebinp.com/Gowv1n2s9CFmn1ik3nAg#hp8ZQovLZCttCuqhUuFKCwuAgTHuCdYvsr0qUpkGcyM=

Installed FastAPI with:
`pip install "fastapi[all]"`

### Operating System

Linux

### Operating System Details

Ubuntu 22.04 running in WSL2.

### FastAPI Version

0.95.0

### Python Version

3.10.6

**Comment by caner-cetin** [comment]
```python
caner in ~/damacansu-backend on master ● ● λ pip install httpx
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: httpx in /home/caner/.local/lib/python3.10/site-packages (0.23.3)
Requirement already satisfied: rfc3986[idna2008]<2,>=1.3 in /home/caner/.local/lib/python3.10/site-packages (from httpx) (1.5.0)
Requirement already satisfied: sniffio in /home/caner/.local/lib/python3.10/site-packages (from httpx) (1.3.0)
Requirement already satisfied: httpcore<0.17.0,>=0.15.0 in /home/caner/.local/lib/python3.10/site-packages (from httpx) (0.16.3)
Requirement already satisfied: certifi in /home/caner/.local/lib/python3.10/site-packages (from httpx) (2022.12.7)
Requirement already satisfied: anyio<5.0,>=3.0 in /home/caner/.local/lib/python3.10/site-packages (from httpcore<0.17.0,>=0.15.0->httpx) (3.6.2)
Requirement already satisfied: h11<0.15,>=0.13 in /home/caner/.local/lib/python3.10/site-packages (from httpcore<0.17.0,>=0.15.0->httpx) (0.14.0)
Requirement already satisfied: idna in /home/caner/.local/lib/python3.10/site-packages (from rfc3986[idna2008]<2,>=1.3->httpx) (3.4)
```

**Comment by caner-cetin** [comment]
...bump. Library utility literally not working.

**Reply by Kludex** [reply]
This is likely to be an issue with your virtual environment.

**Reply by caner-cetin** [reply]
> This is likely to be an issue with your virtual environment.

Bud, I uninstalled Unicorn and FastAPI in my system and re-downloaded within venv and its same result. PyCharm handles it by itself anyways.

**Reply by Kludex** [reply, marked as answer]
Run the following commands please:

python -m pip show httpx
python -m pytest

**Reply by caner-cetin** [reply]
> Run the following commands please:
> 
> python -m pip show httpx python -m pytest

In venv or my system?

**Reply by Kludex** [reply]
Ok, let's go from the beginning.

How are you creating the environment, installing the dependencies and running pytest?

**Reply by caner-cetin** [reply]
> Ok, let's go from the beginning.
> 
> How are you creating the environment, installing the dependencies and running pytest?

@Kludex JetBrains FastAPI preset, I install dependencies like this and never faced an issue until now:
![image](https://user-images.githubusercontent.com/92731060/226588126-09bda039-df97-4b7b-a760-72096ca2bcbd.png)

pytest run:
![image](https://user-images.githubusercontent.com/92731060/226588576-bf3b3a79-1d73-40a1-b5fe-b88e15276f49.png)

like I dont even have to dive into venv in the most time.
